playbooks/common/openshift-cluster/upgrades/etcd/backup.yml

```diff
@@ -49,6 +49,13 @@
     command: >
       etcdctl backup --data-dir={{ openshift.etcd.etcd_data_dir }}
       --backup-dir={{ openshift.common.data_dir }}/etcd-backup-{{ backup_tag | default('') }}{{ timestamp }}
+    when: not openshift.common.is_containerized | bool
+
+  - name: Generate etcd backup
+    command: >
+      docker exec etcd_container etcdctl backup --data-dir={{ openshift.etcd.etcd_data_dir }}
+      --backup-dir={{ openshift.common.data_dir }}/etcd-backup-{{ backup_tag | default('') }}{{ timestamp }}
+    when: openshift.common.is_containerized | bool
```

Comment:
Instead of two different tasks, it might make sense to set the etcdctl command using a variable:

```
vars:
  ...
  etcdctl_command: "{{ 'etcdctl' if not openshift.common.is_containerized else 'docker exec etcd_container etcdctl' }}"
tasks:
  ...
  - name Generate etcd backup
    command: >
       {{ etcdctl_command }} backup --data-dir={{ openshift.etcd.etcd_data_dir }}
        --backup-dir={{ openshift.common.data_dir }}/etcd-backup-{{ backup_tag | default('') }}{{ timestamp }}